### PR TITLE
UIEXPMGR-60 Third pane view - Download and re-send actions

### PR DESCRIPTION
**Pull request by usavkov-epam**
## Purpose

https://issues.folio.org/browse/UIEXPMGR-60
## Approach

- Add `Re-send` and `Download` buttons in Organizations job details action menu;
- Add hook to retrieve EDI job configs and adjust the display of `Re-run` menu item based on configs;
- Add unit tests.
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by usavkov-epam** [comment]
@folio-org/fe-tl-reviewers review please

File: src/ExportEdiJobs/ExportEdiJobDetailsActionMenu/ExportEdiJobDetailsActionMenu.js [hunk 1]
```diff
@@ -3,25 +3,47 @@ import PropTypes from 'prop-types';
...
-  const showCallout = useShowCallout();
+
+  const {
+    name,
+    fileNames,
+    status,
+  } = exportJob;
+
+  const canBeResendOrDownload = (
+    status === EXPORT_JOB_STATUSES.successful || (status === EXPORT_JOB_STATUSES.failed && fileNames?.length)
```
Comment (NikitaSedyx) [inline src/ExportEdiJobs/ExportEdiJobDetailsActionMenu/ExportEdiJobDetailsActionMenu.js:45]: does it mean just fileNames?.length > 0?

File: src/ExportEdiJobs/ExportEdiJobDetailsActionMenu/ExportEdiJobDetailsActionMenu.js [hunk 2]
```diff
@@ -3,25 +3,47 @@ import PropTypes from 'prop-types';
...
   const { navigateToEdiJobDetails } = useNavigation();
-  const showCallout = useShowCallout();
+
+  const {
+    name,
+    fileNames,
+    status,
+  } = exportJob;
+
+  const canBeResendOrDownload = (
```
Comment (NikitaSedyx) [inline src/ExportEdiJobs/ExportEdiJobDetailsActionMenu/ExportEdiJobDetailsActionMenu.js:44]: Resent

Comment (NikitaSedyx) [reply src/ExportEdiJobs/ExportEdiJobDetailsActionMenu/ExportEdiJobDetailsActionMenu.js:44]: but I think it's better to have hasFiles -> that you don't need to have const and can be simply used files?.length > 0 && (<>

File: src/ExportEdiJobs/ExportEdiJobDetailsActionMenu/ExportEdiJobDetailsActionMenu.js (hunk 1 above)
Comment (usavkov-epam) [reply src/ExportEdiJobs/ExportEdiJobDetailsActionMenu/ExportEdiJobDetailsActionMenu.js]: It looks like it, thanks)

File: src/ExportEdiJobs/ExportEdiJobDetailsActionMenu/ExportEdiJobDetailsActionMenu.js (hunk 2 above)
Comment (usavkov-epam) [reply src/ExportEdiJobs/ExportEdiJobDetailsActionMenu/ExportEdiJobDetailsActionMenu.js]: updated

File: src/ExportEdiJobs/ExportEdiJobDetailsActionMenu/ExportEdiJobDetailsActionMenu.test.js [hunk 3]
```diff
@@ -64,4 +82,45 @@ describe('ExportEdiJobDetailsActionMenu', () => {
...
+  });
+
+  it('should handle export job download error', async () => {
+    kyGetMock.mockReturnValue({ blob: () => Promise.reject() });
+
+    renderExportEdiJobDetailsActionMenu();
+
+    await act(async () => user.click(screen.getByTestId('job-action-download')));
+
+    expect(showCallout).toHaveBeenCalled();
```
Comment (BogdanDenis) [inline src/ExportEdiJobs/ExportEdiJobDetailsActionMenu/ExportEdiJobDetailsActionMenu.test.js:106]: Maybe it's also worth checking that it was called with correct message? And same for re-send job

**Review by BogdanDenis — CHANGES_REQUESTED** [review]

File: src/ExportEdiJobs/ExportEdiJobDetailsActionMenu/ExportEdiJobDetailsActionMenu.test.js (hunk 3 above)
Comment (usavkov-epam) [reply src/ExportEdiJobs/ExportEdiJobDetailsActionMenu/ExportEdiJobDetailsActionMenu.test.js]: Yeah, makes sense, thanks

**Review by NikitaSedyx — APPROVED** [review]

**Review by BogdanDenis — APPROVED** [review]

**Pull request merged**